File: chromosom.cpp
using namespace std;

#include "Chromosom.h"

Chromosom::Chromosom()
{
   this->x1 = 0;
   this->x2 = 0;
}

Chromosom::Chromosom(double x1, double x2)
{
    this->x1 = x1;
    this->x2 = x2;
}

Chromosom::Chromosom(const Chromosom& chromosom)
{
    this->x1 = chromosom.x1;
    this->x2 = chromosom.x2;
}

Chromosom::~Chromosom()
{
    ;
}

double Chromosom::pobierzX1()
{
    return this->x1;
}

double Chromosom::pobierzX2()
{
    return this->x2;
}

File: Chromosom.py
"""
Klasa przechowuje chromosom skladajacy sie z dwoch genow typu double. Udostepnia medtody pozwalajace pobierac wartosci genow
"""
class Chromosom(object):

	"""
	Konstruktor przypisujacy chromosomowi zadane wartosci
	"""
	def __init__(self, genX1, genX2):
		self.genX1 = genX1
		self.genX2 = genX2

	"""
	Oblicza funkcje przystosowania dla funkcji celu na podstawie chromosomu
	"""
	@property
	def obliczPrzystosowanie(self):
		return -1 * ((self.genX1 + 2*self.genX2 - 7)*(self.genX1 + 2*self.genX2 - 7) + (2*self.genX1 + self.genX2 -5 )*(2*self.genX1 + self.genX2 -5 ))
